gui/wxpython/datacatalog/tree.py

```diff
@@ -1838,30 +1860,30 @@ def SwitchMapset(self, grassdb, location, mapset, show_confirmation=False):
         """
         Switch to location and mapset interactively.
         """
-        if can_switch_mapset_interactive(self, grassdb, location, mapset):
```

Comment:
What is the reason for moving this if out of the function?

Reply:
The reason is that we display a layer only if a switching is successful. I had another idea how to solve this issue - make a optional argument display_layer for SwitchMapset method. But finally I realized that SwitchMapset is something not much related to displaying layer.
Now code also seems to me a bit clearer because SwitchMapset really switch in all cases.
Making a display_layer argument would be much shorter in code but I think not much meaningful in terms of logic.
Or we could return True or False from SwitchMapset but I am not sure If this function should return something, It is more of procedure. What do you think @petrasovaa ?

Reply:
I see. I would be in favor of returning true/false since we need to use can_switch_mapset_interactive anyway every time we want to switch it, so it may be better have it in the SwitchMapset method.